commerce_coordinator/apps/iap/views.py

```diff
@@ -171,3 +185,114 @@ def post(self, request: Request) -> Response:
                 {"error": message},
                 status=status.HTTP_400_BAD_REQUEST,
             )
+
+
+class IOSRefundView(APIView):
+    """
+    Create refunds for orders refunded by apple
+    """
+
+    apple_cert_file_path = "commerce_coordinator/apps/iap/AppleRootCA-G3.cer"
+
+    def post(self, request):
+        """
+        IOS refund view to receive refund webhook notifications from apple
+        """
+        notification = ios_validator.parse(
+            request.body, apple_root_cert_path=self.apple_cert_file_path
+        )
+        notification_type = notification["notificationType"]
+        logger.info(
+            "Received notification from apple with notification type: "
+            f"{notification_type}"
+        )
+        if notification_type == "REFUND":
+            transaction = notification["data"]["signedTransactionInfo"]
+            refund: Refund = {
+                "id": transaction["originalTransactionId"],
+                "created": transaction["revocationDate"],
+                "amount": convert_localized_price_to_ct_cent_amount(
+                    amount=transaction["price"],
+                    currency_code=transaction["currency"],
+                    # price received from notification is scaled by 1000
+                    # Ref: https://developer.apple.com/documentation/appstoreserverapi/jwstransactiondecodedpayload
+                    exponent=3,
+                ),
+                "currency": transaction["currency"],
+                "status": "succeeded",
+            }
+            payment_refunded_signal.send_robust(
+                sender=self.__class__,
+                payment_interface=EDX_IOS_IAP_PAYMENT_INTERFACE_NAME,
+                refund=refund,
+            )
+        else:
+            logger.info(
+                "Ignoring notification from apple since we are only expecting "
+                "refund notifications"
+            )
+
+        return Response(status=status.HTTP_200_OK)
+
+
+class AndroidRefundView(APIView):
+    """
+    Create refunds for orders refunded by google
+    """
+
+    processor_name = "android_iap"
+    timeout = 30
+
+    def get(self, request):
```

Comment:
If we are going to add a cron job that will execute this endpoint, then we should restrict it to AdminUser only. Alternatively, we can change this to a utility method and call it from the management command's handle function.

Also, if we find a way to make Android call a WH event that we can listen to on this endpoint, then this will get changed to a POST method.